lib/flipper/adapters/moneta.rb

```diff
@@ -0,0 +1,160 @@
+require 'moneta'
+
+module Flipper
+  module Adapters
+    class Moneta
+      include ::Flipper::Adapter
+
+      FEATURES_KEY = :features
+
+      # Public: The name of the adapter.
+      attr_reader :name
+
+      # Public
+      def initialize(moneta)
+        @moneta = moneta
+        @name = :moneta
+      end
+
+      def features
+        read_feature_keys
+      end
+
+      # Public: Adds a feature to the set of known features.
+      def add(feature)
+        set_add(FEATURES_KEY, feature.key)
+        true
+      end
+
+      # Public: Removes a feature from the set of known features and clears
+      # all the values for the feature.
+      def remove(feature)
+        set_delete(FEATURES_KEY, feature.key)
+        clear(feature)
+        true
+      end
+
+      # Public: Clears all the gate values for a feature.
+      def clear(feature)
+        feature.gates.each do |gate|
+          delete key(feature, gate)
```

Comment:
Each of these operations will require a network call for any networked data store, correct? 

I was picturing everything we need for a feature stored in a single key/value (e.g. `"feature/#{feature.key}"` with value serialized by marshal or json or whatever moneta supports). It would increase the number of possible network calls on enable/disable/etc (read + modify value + write vs single write as you currently have it), but would reduce the number of keys to fetch when preloading (1 per feature instead of N per feature where N is the number of gates). 

I've tried to optimize reads (get, get_multi, get_all) in flipper since those are the hotter path (than enabled/disable/clear). How do you feel about switching this to a single key per feature? Any downsides I haven't considered? Only thing I can think of is perhaps concurrent writes, but we could always use Moneta's locking for that if necessary down the road.

Reply:
That's a really great idea!  Definitely makes sense to be optimizing for reads.  Been busy past few weeks with some stuff, but going to spike on that and see how it goes. Off top of my head can't think of any other downsides, but will put more thought into it as I'm jamming